asset-drained/src/agent.js

```diff
@@ -1,200 +1,258 @@
-const { Finding, FindingSeverity, FindingType, ethers, getEthersProvider } = require("forta-agent");
+const { Finding, FindingSeverity, FindingType, ethers, getEthersProvider, Label, EntityType } = require("forta-agent");
 const { MulticallProvider, MulticallContract } = require("forta-agent-tools");
 const LRU = require("lru-cache");
 
 const { hashCode, getAddressType, getAssetSymbol, TOKEN_ABI } = require("./helper");
 const AddressType = require("./address-type");
+const { PersistenceHelper } = require("./persistence.helper");
 
 const ZERO = ethers.constants.Zero;
 const ERC20_TRANSFER_EVENT = "event Transfer(address indexed from, address indexed to, uint256 value)";
 
+const DATABASE_URL = "https://research.forta.network/database/bot/";
+
+let chainId;
+
+const ASSET_DRAINED_TXN_KEY = "nm-asset-drained-bot-key";
+const ALL_TRANSFERS_KEY = "nm-all-transfers-bot-key";
+
+let assetDrainedTransactions = 0;
+let totalTransferTransactions = 0;
+
 const ethcallProvider = new MulticallProvider(getEthersProvider());
 
 const cachedAddresses = new LRU({ max: 100_000 });
 const cachedAssetSymbols = new LRU({ max: 100_000 });
 
 let transfersObj = {};
 
-const initialize = async () => {
-  await ethcallProvider.init();
+const provideInitialize = (provider, persistenceHelper, assetDrainedTxnKey, allTransfersKey) => {
+  return async () => {
+    await ethcallProvider.init();
+
+    chainId = (await provider.getNetwork()).chainId.toString();
+    assetDrainedTransactions = await persistenceHelper.load(
+      assetDrainedTransactions,
+      assetDrainedTxnKey.concat("-", chainId)
+    );
+    totalTransferTransactions = await persistenceHelper.load(
+      totalTransferTransactions,
+      allTransfersKey.concat("-", chainId)
+    );
```

Comment:
```suggestion
    assetDrainedTransactions = await persistenceHelper.load(
      assetDrainedTxnKey.concat("-", chainId)
    );
    totalTransferTransactions = await persistenceHelper.load(
      allTransfersKey.concat("-", chainId)
    );
```

Value doesn't need to be passed.